### PR TITLE
Completed CYF-hotel-react assignment

### PR DESCRIPTION
**Pull request by kiyaGu**
I have completed the assignment and also deployed the app to Heroku you can check it out here https://guarded-escarpment-78547.herokuapp.com/. I have modified the file structure to include a node(express) server file. So, the initial react file is moved to ui-client folder.

### CODE REVIEW

**Comment by Amwam** [comment]
@kiyaGu This is hard to review, as you've committed the node_modules folder.
I think you need to add it to you gitignore file [https://git-scm.com/docs/gitignore](https://git-scm.com/docs/gitignore). Github is refusing to show all the files as there are so many.

File: ui-client/src/components/ResultTableRows.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      })
+    }
+  }
+  
+  render(){ 
+    return (
+      <tr 
+        key ={this.props.index}
+        /*since the key is not accessible for use we pass the index to identify each row from refs  */
+        onClick ={this.rowSelected.bind(this,this.props.index)}
```
Comment (Amwam) [inline ui-client/src/components/ResultTableRows.js]: Using bind does work, another way would be to write this as:
```jsx
{() => this.rowSelected(this.props.index)}
```

(This is more personal  preference)

File: ui-client/src/components/Results.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import ResultTableHeaders from './ResultTableHeaders';
+import ResultTableRows from './ResultTableRows';
+
+//to claculate the number of days that a customer has reserved/booked
+const calculateDateDifference = (reservations)=>{
+   reservations.map(reservation =>{
+     let checkOutDate = moment(reservation.checkOutDate);
+     let checkInDate = moment(reservation.checkInDate);
+     //calculate the difference and include the result,totalDays, to the booking object
+     return reservation.totalDays = checkOutDate.diff(checkInDate,'days')
```
Comment (Amwam) [inline ui-client/src/components/Results.js]: This is quite confusing.
If you're just trying to add a value to the `reservation` object, don't bother returning anything.

File: ui-client/src/containers/Bookings.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    searchByCustomerFirstName = (e) => {
+      e.preventDefault();
+      let customerName = this.state.customerName;
+      if(customerName === null || customerName === ''){
+        alert("PLEASE, you need to specify the First name of the customer before you click search!!!");
+      }
+      else{
+        const reservations = this.props.reservations
+          .filter(reservation => (
+              reservation.firstname.toString().toLowerCase() === customerName.toLowerCase()
```
Comment (Amwam) [inline ui-client/src/containers/Bookings.js]: Isn't firstname a string already?

File: ui-client/src/App.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                    Failed to load resource:
+                </h1>
+                <div className="error-details">
+                    Sorry, the server responded with a status of 404 &#40;Not Found&#41;
+                </div>
+            </div>
+            <Footer />
+          </div>)
+      //if the API resource request hasn't yet responded with 200 or ok
+      if(this.state.loading)
```
[the remark this answers is not in the document]
Comment (kiyaGu) [reply ui-client/src/App.js]: I already got Eslint and when I made my code if ({this.state.loading}) Eslint says [js] ':' expected that was why I avoided

**Pull request closed without merging**